R/expect-named.R

```diff
@@ -41,19 +41,42 @@ expect_named <- function(
       sprintf("%s does not have names.", act$lab)
     )
   } else {
-    exp_names <- normalise_names(expected, ignore.order, ignore.case)
+    exp <- quasi_label(enquo(expected), arg = "expected")
+    exp$val <- normalise_names(exp$val, ignore.order, ignore.case)
     act$names <- normalise_names(act$names, ignore.order, ignore.case)
 
-    expect(
-      identical(act$names, exp_names),
-      sprintf(
-        "Names of %s (%s) don't match %s",
-        act$lab,
-        paste0("'", act$names, "'", collapse = ", "),
-        paste0("'", exp_names, "'", collapse = ", ")
-      ),
-      info = info
-    )
+    if (ignore.order) {
+      act_miss <- unique(act$names[!act$names %in% exp$val])
+      exp_miss <- unique(exp$val[!exp$val %in% act$names])
+
+      expect(
```

Comment:
This `expect()` block is copied and modified slightly from `expect_setequal()` to include `"Names of "`.

Reply:
Could you explore extracting out a single helper function that we could use here, in `expect_setequal()` and `expect_mapequal()`?  A expectation helper should end in `_` and have an argument `trace_env = caller_env()` that is passed to every use of `fail()`.